cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -128,8 +128,9 @@ class SimpleArrayMixinCalculators
 
     using value_type = typename internal_types::value_type;
 
-    value_type min(value_type initial = std::numeric_limits<value_type>::max()) const
```

Comment:
the old function increases the difficulty of implementing the method in WrapSimpleArrayPlex.cpp. I checked the usage cases in modmesh, I don't think we need to have an argument here. We can add the argument in the future if needed.

Reply:
The default argument is irrelevant in most use cases.  When we need it, it's also trivial to add an argument back through overloading or something.

The change here is benign and minor.